### DISCUSSION TITLE
Can I used Hydra as a federated identity provider?

### DISCUSSION THREAD

**Discussion by osbornk** [ory/kratos#3620] [Q&A] [answered]
This is mostly for mobile, but also applies to web. We are looking to replace our token/session management for a first-party mobile app. It has a bunch of problems. So, Kratos looks promising for our use case.

But our identity store is back in our monolith. It does a bunch of complicated logic, including optionally redirecting a user to various SAML providers based on their email domain. Even for the username/password case, it is too much to migrate the entire identity store to Kratos at this point in time. So, I would really just want to use Kratos for the session management, not authentication. But I do want to unify things. Ultimately, whether the user has logged in via username/password or SAML, I want the same session management.

But Kratos can call OIDC and Hydra is an OIDC provider. One idea I had is to hide all the SAML and other complex auth logic as the login URL for Hydra. And then use the social login features of Kratos. So, the first-party application has the Kratos session management. But it uses OIDC behind the scenes.

Or am I thinking of it completely wrong here?

Going directly to Hydra is an option, but I don't plan on allowing third-party access so that feels wrong to me.

**Comment by kmherrmann** [comment, marked as answer]
I think you described the solution picture accurately:
- your apps speak OIDC/OAuth2 with Ory OAuth2 (based on hydra), or directly use Ory Identities (based on Kratos) to cut out complexity in 1P apps
- Identities uses OIDC to log in users via social providers (or even a legacy system if that speaks OIDC)
- Ory Network (Identities) supports domain-bound SSO connections (Scale & Enterprise tiers). If you have to connect SAML providers, we have a partnership to facilitate that, too. Open Source doesn't have the domain bound Enterprise SSO, though.
